File: Sorting Algorithms/quick_sort.cpp
#include <iostream>

using namespace std;

void swap(int *a, int *b)
{
    int temp = *a;
    *a = *b;
    *b = temp;
}

int partition(int arr[], int low, int high)
{
    int pivot = arr[high]; // pivot element
    int i = (low - 1);     // Index of smaller element

    for (int j = low; j <= high - 1; j++)
    {
        // If current element is smaller than or
        // equal to pivot
        if (arr[j] <= pivot)
        {
            i++; // increment index of smaller element
            swap(&arr[i], &arr[j]);
        }
    }
    swap(&arr[i + 1], &arr[high]);
    return (i + 1);
}

void sorter(int arr[], int low, int high)
{
    if (low < high)
    {
        /* pi is partitioning index, arr[p] is now 
           at right place */
        int pi = partition(arr, low, high);

        // Separately sort elements before
        // partition and after partition
        sorter(arr, low, pi - 1);
        sorter(arr, pi + 1, high);
    }
}

void quickSort(int arr[], int n)
{
    sorter(arr, 0, n - 1);
}

int main(int argc, char const *argv[])
{
    int N;
    cin >> N;
    int arr[N], index;
    for (index = 0; index < N; index++)
    {
        cin >> arr[index];
    }
    quickSort(arr, N);
    // printing the array elements
    for (index = 0; index < N; index++)
    {
        cout << arr[index] << " ";
    }
    return 0;
}

File: Sorting Algorithms/Quick_Sort.py
def partition(Test_list, low, high):
    i = low-1
    pivot = Test_list[high]
    for j in range(low, high):
        if Test_list[j] < pivot:
            i += 1
            Test_list[i], Test_list[j] = Test_list[j], Test_list[i]
    Test_list[i+1], Test_list[high] = Test_list[high], Test_list[i+1]
    return (i+1)


def Quick_Sorting(Test_list, low, high):
    if len(Test_list) == 1:
        return Test_list
    if low < high:
        Part = partition(Test_list, low, high)
        Quick_Sorting(Test_list, low, Part-1)
        Quick_Sorting(Test_list, Part+1, high)

Test_list = [11, 35, 12, 36, 10, 6, 25]
Quick_Sorting(Test_list, 0, len(Test_list)-1)
print(Test_list)